### DISCUSSION TITLE
S3 Client not Exposing HeadBucket.

### DISCUSSION THREAD

**Discussion by omamoo** [aws/aws-sdk-net#3175] [Q&A] [answered]
Hello all,

I'm writing a code that uploading content to S3 bucket that resides at different account, i.e. I have a cross account scenario of uploading content to S3.

While writing the code I used `GetBucketLocation` API call to determine the bucket location, however on cross account scenario the `GetBucketLocation` fails since I'm not the bucket owner, in addition, I saw on AWS documentation that their recommendation is to use `HeadBucket` API call, however the SDK does not expose this API call and defined this call as `internal`. 

Here is a snapshot from the SDK code:
```
internal virtual Task<HeadBucketResponse> HeadBucketAsync(HeadBucketRequest request, System.Threading.CancellationToken cancellationToken = default(CancellationToken))
        {
            var options = new InvokeOptions();
            options.RequestMarshaller = HeadBucketRequestMarshaller.Instance;
            options.ResponseUnmarshaller = HeadBucketResponseUnmarshaller.Instance;

            return InvokeAsync<HeadBucketResponse>(request, options, cancellationToken);
        }
``` 

Can we make this public for use?

**Comment by dscpinheiro** [comment]
I found this issue from a few years ago, mentioning we didn't make the operation public because there's a separate method (`AmazonS3Util.DoesS3BucketExist`) that can be used to verify a bucket exists: https://github.com/aws/aws-sdk-net/issues/616

That doesn't really help your case (or if you're interested in any of the headers `HeadBucket` returns). I'll convert this to an issue so we can track it.

**Reply by omamoo** [reply]
@dscpinheiro 
In my case, it is does not meet my requirements, i interested with the bucket location header. 
Since AWS GetBucketLocation operates in different way, and does not operate well on cross account requests. 

It's something that can be done? If you can make this as a feature request it will be perfect.

Thanks.

**Reply by dscpinheiro** [reply, marked as answer]
Yes, I opened a new issue to make the `HeadBucket` operation public in our S3 client: #3178
